RecoLocalMuon/GEMRecHit/src/ME0SegAlgoMM.cc

```diff
@@ -0,0 +1,556 @@
+      // the clusterHits one below but we leave it for security and backward 
+      // comparison reasons 
+      rechits_clusters = this->chainHits( rechits );
+    }
+    else{
+      // it uses X,Y information + configurable parameters
+      rechits_clusters = this->clusterHits(rechits );
+    }
+    // loop over the found clusters:
+    for(auto sub_rechits = rechits_clusters.begin(); sub_rechits !=  rechits_clusters.end(); ++sub_rechits ) {
+      // clear the buffer for the subset of segments:
+      segments_temp.clear();
+      // build the subset of segments:
+      segments_temp = this->buildSegments( (*sub_rechits) );
+      // add the found subset of segments to the collection of all segments in this chamber:
+      segments.insert( segments.end(), segments_temp.begin(), segments_temp.end() );
+    }
+  
+
+    return segments;
+  }
+  else {
+    segments = this->buildSegments(rechits);
+    return segments;
+  }
+}
+
+
+// ********************************************************************;
+ME0SegAlgoMM::ProtoSegments 
+ME0SegAlgoMM::clusterHits(const EnsambleHitContainer & rechits) {
+
+  ProtoSegments rechits_clusters; // this is a collection of groups of rechits
+  //   const float dXclus_box_cut       = 4.; // seems to work reasonably 070116
+  //   const float dYclus_box_cut       = 8.; // seems to work reasonably 070116
+
+  float dXclus_box = 0.0;
+  float dYclus_box = 0.0;
+
+  EnsambleHitContainer temp;
+  ProtoSegments seeds;
+
+  std::vector<float> running_meanX;
+  std::vector<float> running_meanY;
+
+  std::vector<float> seed_minX;
+  std::vector<float> seed_maxX;
+  std::vector<float> seed_minY;
+  std::vector<float> seed_maxY;
+
+   // split rechits into subvectors and return vector of vectors:
+  // Loop over rechits 
+  // Create one seed per hit
+  for(unsigned int i = 0; i < rechits.size(); ++i) {
+    temp.clear();
+    temp.push_back(rechits[i]);
+    seeds.push_back(temp);
+
+    // First added hit in seed defines the mean to which the next hit is compared
+    // for this seed.
+
+    running_meanX.push_back( rechits[i]->localPosition().x() );
+    running_meanY.push_back( rechits[i]->localPosition().y() );
+	
+    // set min/max X and Y for box containing the hits in the precluster:
+    seed_minX.push_back( rechits[i]->localPosition().x() );
+    seed_maxX.push_back( rechits[i]->localPosition().x() );
+    seed_minY.push_back( rechits[i]->localPosition().y() );
+    seed_maxY.push_back( rechits[i]->localPosition().y() );
+  }
+    
+  // merge clusters that are too close
+  // measure distance between final "running mean"
+  for(size_t NNN = 0; NNN < seeds.size(); ++NNN) {
+    for(size_t MMM = NNN+1; MMM < seeds.size(); ++MMM) {
+      if(running_meanX[MMM] == 999999. || running_meanX[NNN] == 999999. ) {
+	LogDebug("ME0Segment|ME0") << "ME0SegAlgoMM::clusterHits: Warning: Skipping used seeds, this should happen - inform developers!";
```

Comment:
I think this message should say "this should NOT happen". Also, please delete the cout comment.